server/database/index.js

```diff
@@ -1,137 +1,72 @@
-    isbn13: Number,
-    asin: String,
-    language: String,
-    series: {
-      name: String,
-      url: String,
-    },
-  },
-});
-
-const Book = mongoose.model('Book', bookSchema);
-
-const seed = (Model, callback) => {
-  // clean out current database, if any test records clogging up
-  Model.deleteMany({}, async () => {
-    let book;
-    const queries = [];
-    for (let i = 0; i < 100; i += 1) {
-      book = new Model();
-
-      // initiate a bunch of new book info
-      book.id = i;
-      book.title = fake.random.words(3);
-      book.author = fake.name.findName();
-      book.description = fake.lorem.paragraphs();
-      book.ratings = {
-        five: fake.random.number(),
-        four: fake.random.number(),
-        three: fake.random.number(),
-        two: fake.random.number(),
-        one: fake.random.number(),
-      };
-      book.reviews = fake.random.number();
-      book.links = {
-        kindle: fake.internet.url(),
-        amazon: fake.internet.url(),
-        stores: {
-          audible: fake.internet.url(),
-          barnesAndNoble: fake.internet.url(),
-          walmart: fake.internet.url(),
-          apple: fake.internet.url(),
-          google: fake.internet.url(),
-          abebooks: fake.internet.url(),
-          bookDesository: fake.internet.url(),
-          indigo: fake.internet.url(),
-          alibris: fake.internet.url(),
-          betterWorldBooks: fake.internet.url(),
-          indieBound: fake.internet.url(),
-        },
-      };
-      book.type = fake.random.word();
-      book.pages = fake.random.number({ max: 3000 });
-      book.publishDate = fake.date.past();
-      book.publisher = fake.company.companyName();
-      book.metadata = {
-        originalTitle: book.title,
-        isbn: fake.random.number(),
-        isbn13: fake.random.number(),
-        language: 'English',
-      };
-      book.image = 'http://lorempixel.com/480/640/abstract/';
-
-      if (i % 2 === 0) {
-        book.series = {
-          name: fake.random.words(2),
-          url: fake.internet.url(),
-        };
+const findMaxId = (callback) => {
+  bookModel.findOne({})
+    .sort('-id')
+    .exec((err, item) => {
+      if (item) {
+        callback(item.id);
+      } else {
+        callback(0);
       }
+    });
+};
+const add = (newbookData, callback) => {
+  console.log('New Book Data:', newbookData);
+  findMaxId((id) => {
+    // eslint-disable-next-line no-param-reassign
+    newbookData.id = id + 1;
+    const newbook = new bookModel(newbookData);
+    newbook.save((err, result) => {
+      console.log('Add result is', result);
+      callback(err, result);
+    });
+  });
+};
 
-      queries.push(book.save());
+
+const update = (bookData, callback) => {
+  bookModel.findOne({ id: bookData.id }, (err, book) => {
+    if (!err) {
+      book.title = bookData.title;
+      book.save((error, result) => {
+        console.log('Update result is', result);
+        callback(error, result);
+      });
```

Comment:
That's not the right way for updating a document. You assumed that the `title` is the property to be updated, what if it was another property let alone a punch of properties. You might want to consider the `findOneAndUpdate` function & use the whole `req.body` object as a second argument for it.